### ISSUE TITLE
Another example of duplicate packages with different versions using `--universal`

### ISSUE THREAD

**Issue by notatallshaw-gts** [astral-sh/uv#4992]
```
$ uv -V
uv 0.2.24

$ echo -e "pylint>=2.14.5\ndill==0.3.1.1" | uv pip compile --python-version 3.10.6 --universal --annotation-style line - | grep "pylint=="
Resolved 14 packages in 43ms
pylint==2.15.8
pylint==3.2.5
```

My original example had the dill requirement passed in via a constraint, but after cutting out hundreds of requirements and constraints was able to reproduce a minimal example with just requirements.

**Comment by konstin** [comment]
Thanks for the clear reproduction!

`pylint==3.25` has:
```
Requires-Dist: dill >=0.2 ; python_version < "3.11"
Requires-Dist: dill >=0.3.6 ; python_version >= "3.11"
Requires-Dist: dill >=0.3.7 ; python_version >= "3.12"
```

We're solving once for `python_version >= '3.12'` and once for `python_version < '3.11'`, there's at least some part of #4732 here.

**Comment by notatallshaw-gts** [comment]
This appears to be solved in 0.2.35 (https://github.com/astral-sh/uv/pull/5898?) I now get:

```
pylint==2.15.8 ; python_version >= '3.11'
pylint==3.2.6 ; python_version < '3.11'
```
